### ISSUE TITLE
Allow to take ASR as text and compile via LFortran

### ISSUE THREAD

**Issue by certik** [lfortran/lfortran#4687]
Similar to https://github.com/lfortran/lfortran/issues/4686, but for ASR text output. 

So that the following:
```
lfortran a.f90
```
Can also be done using:
```
lfortran a.f90 --show-asr > a.asr
lfortran a.asr
```
And it would be equivalent. Furthermore, we can then take the final ASR before LLVM, and then we need to add a simple way to skip all ASR passes in `lfortran b.asr`, so that we can use `b.asr` to drive directly our LLVM backend. 

This will aid with debugging the LLVM backend, ensuring it works.

**Comment by Pranavchiku** [comment]
Do we not need an ASR parser for this?

**Comment by certik** [comment]
Yes, we need an ASR parser: https://github.com/lfortran/lfortran/issues/2357.
It's not difficult, but it's also not completely a small task.

The other issue is that we should change the syntax to be completely Clojure syntax: https://github.com/lfortran/lfortran/issues/4694.
